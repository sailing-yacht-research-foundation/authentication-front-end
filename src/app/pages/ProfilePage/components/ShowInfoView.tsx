import React from 'react';
import styled from 'styled-components';
import { getUserAttribute } from 'utils/user-utils';
import { Descriptions } from 'antd';

export const ShowInfoView = (props) => {
    const { authUser } = props;
    
    return (
        <Wrapper>
            <Descriptions title="User Info" layout="vertical">
                <Descriptions.Item label="Email">{ getUserAttribute(authUser, 'email') }</Descriptions.Item>
                <Descriptions.Item label="Name">{ getUserAttribute(authUser, 'name') }</Descriptions.Item>
                <Descriptions.Item label="Phone Number">{ getUserAttribute(authUser, 'phone_number') }</Descriptions.Item>
                <Descriptions.Item label="Address">{ getUserAttribute(authUser, 'address') }</Descriptions.Item>
                <Descriptions.Item label="World Sailing Number">{ getUserAttribute(authUser, 'custom:sailing_number') }</Descriptions.Item>
                <Descriptions.Item label="Facebook">{ getUserAttribute(authUser, 'custom:facebook') }</Descriptions.Item>
                <Descriptions.Item label="Instagram">{ getUserAttribute(authUser, 'custom:instagram') }</Descriptions.Item>
                <Descriptions.Item label="Twitter">{ getUserAttribute(authUser, 'custom:twitter') }</Descriptions.Item>
                <Descriptions.Item label="Address">
                { getUserAttribute(authUser, 'address') }
                </Descriptions.Item>
            </Descriptions>
        </Wrapper>
    );
}

const Wrapper = styled.div`
    margin-top: 30px;
    width: 100%;
`