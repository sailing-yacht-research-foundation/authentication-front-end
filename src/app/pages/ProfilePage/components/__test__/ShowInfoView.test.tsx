
import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { ShowInfoView } from '../ShowInfoView';

const shallowRenderer = createRenderer();

describe('<ShowInfoView />', () => {
  it('should render to match the snapshot', () => {
    shallowRenderer.render(<ShowInfoView />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});