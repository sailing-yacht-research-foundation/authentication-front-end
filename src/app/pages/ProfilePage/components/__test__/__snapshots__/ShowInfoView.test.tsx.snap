// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowInfoView /> should render to match the snapshot 1`] = `
<styled.div>
  <Descriptions
    layout="vertical"
    title="User Info"
  >
    <DescriptionsItem
      label="Email"
    />
    <DescriptionsItem
      label="Name"
    />
    <DescriptionsItem
      label="Phone Number"
    />
    <DescriptionsItem
      label="Address"
    />
    <DescriptionsItem
      label="World Sailing Number"
    />
    <DescriptionsItem
      label="Facebook"
    />
    <DescriptionsItem
      label="Instagram"
    />
    <DescriptionsItem
      label="Twitter"
    />
    <DescriptionsItem
      label="Address"
    />
  </Descriptions>
</styled.div>
`;
